src/desktop/components/inquiry_questionnaire/test/views/account.test.ts

```diff
@@ -112,6 +112,86 @@ describe(
       })
     })
 
+    describe("two-factor authentication", () => {
+      beforeEach(() => {
+        view.user.related = sinon.stub().returns({ account: { id: "user-id" } })
+        view.active.set("mode", "login")
+        view.render()
+      })
+
+      it("hides the OTP field initially in login form", () => {
+        view
+          .$(".iq-otp-field")
+          .hasClass("is-hidden")
+          .should.be.true()
+      })
+
+      describe("when OTP is required but missing", () => {
+        beforeEach(() => {
+          view.user.related = sinon
+            .stub()
+            .returns({ account: { id: "user-id-2fa-enabled" } })
+          sinon
+            .stub(Backbone, "sync")
+            .onCall(0)
+            .yieldsTo("error", {
+              responseJSON: {
+                error: "missing two-factor authentication code",
+              },
+            })
+        })
+
+        afterEach(() => {
+          Backbone.sync.restore()
+        })
+
+        it("reveals the OTP field", done => {
+          view.$("button").click()
+
+          setTimeout(() => {
+            view
+              .$(".iq-otp-field")
+              .hasClass("is-hidden")
+              .should.be.false()
```

Comment:
Would have preferred to test this with something more explicit like `.is(:visible).should.be.true()`, but for the life of me couldn't get this working in this mocked browser environment — thus switched to this classname based approach here and L125.